system/Database/MigrationRunner.php

```diff
@@ -817,6 +872,89 @@ public function ensureTable()
         $this->tableChecked = true;
     }
 
+    /**
+     * Ensures that we have created our migration
+     * lock table in the database.
+     *
+     * @return string The lock table name
+     */
+    protected function ensureLockTable(): string
+    {
+        $lockTable = $this->table . '_lock';
+
+        if ($this->lockTableChecked || $this->db->tableExists($lockTable)) {
+            return $lockTable;
+        }
```

Comment:
if `$this->lockTableChecked` is `false` and the lock table exists in db, should you set `$this->lockTableChecked` to `true` so next call won't call db again?

Reply:
You're right. Updated.